Sources/Sentry/SentryNSURLRequestBuilder.m

```diff
@@ -8,25 +9,33 @@
 
 @implementation SentryNSURLRequestBuilder
 
-- (NSURLRequest *)createEnvelopeRequest:(SentryEnvelope *)envelope
-                                    dsn:(SentryDsn *)dsn
-                       didFailWithError:(NSError *_Nullable *_Nullable)error
+- (nullable NSURLRequest *)createEnvelopeRequest:(SentryEnvelope *)envelope
+                                             dsn:(SentryDsn *)dsn
+                                didFailWithError:(NSError *_Nullable *_Nullable)error
 {
-    return [[SentryNSURLRequest alloc]
-        initEnvelopeRequestWithDsn:dsn
-                           andData:[SentrySerialization dataWithEnvelope:envelope]
-                  didFailWithError:error];
+    NSData *data = [SentrySerialization dataWithEnvelope:envelope];
+    if (nil == data) {
+        SENTRY_LOG_ERROR(@"Envelope cannot be converted to data");
+        return nil;
+    }
```

Comment:
`m`: As this is now essential functionality, please add a test for this. We don't have a `SentryNSURLRequestBuilder` test class yet, but I think we should add one.